### ISSUE TITLE
Empty redirects created when using CustomUrls

### ISSUE THREAD

**Issue by Boddlnagg** [modxcms/Redirector#54]
Whenever I create a new resource in one of my collections (see [Collections Extra](https://github.com/modxcms/Collections)), Redirector creates an empty redirect to that resource, i.e. the pattern is empty and the target is [[~id]] where id is the ID of the newly created resource. This happens even though redirector.track_uri_updates is turned off.

I'm using Redirector 2.0.6.

**Comment by bertoost** [comment]
Fixed correctly using the "redirector.track_uri_updates" setting.
Cannot reproduce the empty pattern and the MODX link tag as target. It's also only storing non-dynamic URL's. Maybe some other plugin who break things? I tested with a clean install.

This will be available in the next update btw.

**Comment by Boddlnagg** [comment]
Thanks for the fix! As far as the other issue is concerned: When redirects are created automatically on change of an alias, the target is always a dynamic URL for me (i.e. [[~id]]). I thought this was intended. The empty-pattern-redirects could also have to do with the [Custom Urls](https://rtfm.modx.com/extras/revo/customurls) Extra, which I'm using to rewrite the aliases for resources in my collections. Maybe `OnDocFormSave` CustomUrls generates an alias for me and then Redirector thinks that the URL has changed (but the old one is empty).

**Comment by bertoost** [comment]
Try it with temporary disabling the Custom URLs plugin. When it works by then, you know it ;-)

**Comment by Boddlnagg** [comment]
Yeah, it definitely has to do with CustomUrls. Do you have any idea how to fix this, so the plugins don't interact badly with each other? (I considered changing the plugin priority of one of them, but it's not clear to me which one should run first to avoid any issues.)

**Comment by bertoost** [comment]
Yeah that's definitely the way to fix it I guess.
Probably Redirector needs to run first, since it works with session values for old<>new URL's.

**Comment by Boddlnagg** [comment]
If Redirector runs first, however, the alias change triggered by CustomUrls (which generated the alias based on the title in my case) will not be reflected in the Redirector rule. The optimal order would be:
- Redirector reads old URL
- CustomUrls generates a new one
- Redirector uses new URL as target

I guess this is not possible? Therefore, it's probably better if CustomUrls runs first. Couldn't you still work around the issue with empty patterns by just checking whether the old URL is empty? There is no case where rules with empty patterns make any sense, right? So just skip creating the rule in such a case.

**Comment by bertoost** [comment]
Basically that's what is happening.. Redirector plugin also runs OnDocFormRender to store the current URL. When that one is changed against the new one (at save) it will create a rule.
Therefor I have will investigate together with Custom Urls next days

**Comment by Boddlnagg** [comment]
This is still an issue which is slightly annoying, because I have to delete a bunch of empty redirect rules every once in a while.
